main.go

```diff
@@ -17,20 +17,23 @@ package main
 import (
 	"sync"
 
+	ulog "github.com/tigrisdata/fdb-exporter/util/log"
+
 	server "github.com/tigrisdata/fdb-exporter/http"
 	"github.com/tigrisdata/fdb-exporter/metrics"
 )
 
 func main() {
 	var wg sync.WaitGroup
-	closer := metrics.InitMetrics()
-
-	defer closer.Close()
+	mInfo := metrics.NewMetricInfo()
+	defer func() {
+		ulog.E(mInfo.Closer.Close())
```

Comment:
nice. You could change this slightly so that `mInfo` has a function like this:

```
  func (m metricIntro) Close() {
    ulog.E(mCloser.Close())

} 
```

That way you not exposing internal state and if you need to have other things to close in the metrics struct you have a single place to do that.

Reply:
Great! did it like that